bin/mmdb_utils.py

```diff
@@ -236,35 +256,25 @@ def download_mmdb_database(self, license_key):
         # NOTE - Please visit GitHub page (https://github.com/VatsalJagani/Splunk-App-Auto-Update-MaxMind-Database), if you are developer and want to help improving this App in anyways
 
         logger.debug("Downloading the MaxMind DB file.")
-        r = requests.get(MaxMindDatabaseDownloadLink.format(license_key), allow_redirects=True, proxies=proxies)
-        
+        r = requests.get(MaxMindDatabaseDownloadLink, auth=(account_id, license_key), allow_redirects=True, proxies=proxies)
+
         if r.status_code == 200:
-            open(DB_TEMP_DOWNLOAD, 'wb').write(r.content)
+            with open(DB_TEMP_DOWNLOAD, 'wb') as fp:
+                fp.write(r.content)
+
+            downloaded_file = os.path.join(DB_DIR_TEMP_PATH, 'GeoLite2-City.mmdb')
             try:
                 # Extract the downloaded file
-                tar = tarfile.open(DB_TEMP_DOWNLOAD, "r:gz")
-                tar.extractall(path=DB_DIR_TEMP_PATH)
-                tar.close()
+                with gzip.open(DB_TEMP_DOWNLOAD, 'rb') as s_file, open(downloaded_file, 'wb') as d_file:
+                    shutil.copyfileobj(s_file, d_file, 65536)
             except Exception as e:
-                msg = "Unable to untar downloaded MaxMind database. {}".format(e)
+                msg = "Unable to extract downloaded MaxMind database. {}".format(e)
                 logger.exception(msg)
+                raise e
 
-            try:
```

Comment:
Don't this required anymore?

Reply:
Nope. File will be downloaded as "temp_file.gz" and when we unzip it will directly be "GeoLite2-City.mmdb" file.